### PR TITLE
15 12 11 aem6.1 with mongo db 3.0 and wired tiger

### PR DESCRIPTION
**Pull request by ncoble**
I have been a racker since 2007.  I am currently a product engineer on the Rackspace Digital team.  My previous positions have included front line support, supporting the Critical Application Services infrastrucutre, as well as leading that team.  Outside of Rackspace I enjoy spending time with my wife, Tiffany, and two daughters Autumn and Meadow.
www.linkedin.com/in/NathanCoble
Accept any changes made by technical writer staff
12-11-2015
yes to answer comments

### CODE REVIEW

File: _posts/2015-12-11-AEM6.1-With-MongoDB-3.0-And-WiredTiger.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+    enabled: true
+  engine:"WiredTiger"
+#  mmapv1:
+#  wiredTiger:
+```
+Start MongoDB.
+```sh
+$ systemctl start mongod
+```
+Create a MongoDB user for AEM. (For production create your user admin and root user as well.)
```
Comment (catlook) [inline _posts/2015-12-11-AEM6.1-With-MongoDB-3.0-And-WiredTiger.md]: ... For production, create your  ...

File: _posts/2015-12-11-AEM6.1-With-MongoDB-3.0-And-WiredTiger.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+```sh
+security:
+  authorization: enabled
+```
+```sh
+systemctl restart mongod
+```
+
+### Installing AEM 6.1
+[Download Java][java1] and install.
```
Comment (catlook) [inline _posts/2015-12-11-AEM6.1-With-MongoDB-3.0-And-WiredTiger.md]: [Download Java][java1], and install.

File: _posts/2015-12-11-AEM6.1-With-MongoDB-3.0-And-WiredTiger.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+```sh
+$ rpm -Uvh jdk-8u65-linux-x64.rpm
+```
+
+Unpack the AEM jar.
+```sh
+$ java -jar AEM_6.1_Quickstart.jar -unpack
+```
+
+Change the persistence mode and add mongo options to AEM start script.
```
Comment (catlook) [inline _posts/2015-12-11-AEM6.1-With-MongoDB-3.0-And-WiredTiger.md]: ... persistence mode, and add mongo ...

File: _posts/2015-12-11-AEM6.1-With-MongoDB-3.0-And-WiredTiger.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+$ systemctl restart mongod
+```
+Log in to mongo as user admin, and create a new aem user with MongoDB-CR credentials.
+```sh
+> use aem
+switched to db aem
+> db.createUser({ user:"aem6", pwd: "verysecure", roles:["readWrite"]})
+Successfully added user: { "user" : "aem6", "roles" : [ "readWrite" ] }
+```
+Check the user was created with the MONGODB-CR authentication type.
```
Comment (catlook) [inline _posts/2015-12-11-AEM6.1-With-MongoDB-3.0-And-WiredTiger.md]: Check that the user was created ...

File: _posts/2015-12-11-AEM6.1-With-MongoDB-3.0-And-WiredTiger.md [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+```sh
+> use admin
+switched to db admin
+> db.system.users.find({user: 'aem6'})
+{ "_id" : "aem.aem6", "user" : "aem6", "db" : "aem", "credentials" : { "SCRAM-SHA-1" : { "iterationCount" : 10000, "salt" : "DKat2JSX0wTKgv8RQnEhag==", "storedKey" : "LSEytpYhXtsDLTRN0jqc/dxVt24=", "serverKey" : "PnnHurinmY4hFy8eKfaLaXwNRv8=" } }, "roles" : [ { "role" : "readWrite", "db" : "aem" } ] }
+```
+That did not work the "credentials" are still showing "SCRAM-SHA-1". MongoDB is not creating users with the authentication specified. [MongoDB gives us a workaround.][mongo2]
+
+Change the auth schema in MongoDB as suggested above. Start MongoDB without authentication, then login and update the schema.
```
Comment (catlook) [inline _posts/2015-12-11-AEM6.1-With-MongoDB-3.0-And-WiredTiger.md]: ... authentication, and then login ...

File: _posts/2015-12-11-AEM6.1-With-MongoDB-3.0-And-WiredTiger.md [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+WriteResult({ "nMatched" : 1, "nUpserted" : 0, "nModified" : 1 })
+```
+Create new user for aem (aem-cr).
+```sh
+> use aem6
+switched to db aem6
+> db.createUser({ user: "aem-cr", pwd: "verysecure", roles:["readWrite"]})
+Successfully added user: { "user" : "aem-cr", "roles" : [ "readWrite" ] }
+```
+Check the user was added with correctly.
```
Comment (catlook) [inline _posts/2015-12-11-AEM6.1-With-MongoDB-3.0-And-WiredTiger.md]: Check that the user ...

File: _posts/2015-12-11-AEM6.1-With-MongoDB-3.0-And-WiredTiger.md [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+{ "_id" : "aem6.aem-cr", "user" : "aem-cr", "db" : "aem6", "credentials" : { "MONGODB-CR" : "86ccfed007e6efa1efa49f26dd2aa08a" }, "roles" : [ { "role" : "readWrite", "db" : "aem6" } ] }
+```
+Looks good! The "aem-cr" user has credentials for MONGODB-CR.
+
+Next, the AEM start script needs to be updated to specify the authentication type we want for Mongo. Update your Java options with your authMechanism.
+```sh
+CQ_JVM_OPTS='-server -Xmx20g -XX:MaxPermSize=256M -Djava.awt.headless=true -Doak.mongo.uri=mongodb://aem-cr:verysecure@172.31.30.233/aem6?authsource=aem6&authMechanism=MONGODB-CR -Doak.mongo.db=aem6
+```
+Start AEM and verify there are no errors.
+In MongoDB we should now be able to see the collections.
```
Comment (catlook) [inline _posts/2015-12-11-AEM6.1-With-MongoDB-3.0-And-WiredTiger.md]: In MongoDB, we should ...

File: _posts/2015-12-11-AEM6.1-With-MongoDB-3.0-And-WiredTiger.md [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+system.indexes
+```
+There it is AEM 6.1 using MongoDB 3.0 and WiredTiger!
+
+
+   [adobe1]: <https://docs.adobe.com/docs/en/aem/6-1/deploy/technical-requirements.html>
+   [or1]: <https://objectrocket.com/blog/company/mongodb-wiredtiger>
+   [java1]: <http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html>
+   [mongo1]: <https://docs.mongodb.org/manual/core/authentication/>
+   [mongo2]: <https://jira.mongodb.org/browse/SERVER-17459>
```
Comment (catlook) [inline _posts/2015-12-11-AEM6.1-With-MongoDB-3.0-And-WiredTiger.md]: [ed note: nice blog!  Let me know if you have questions about the suggested edits - Cat ]

**Comment by ncoble** [comment]
Thanks! Edits committed.

**Comment by ktbartholomew** [comment]
@ncoble sorry I missed this on Friday. If you'll update the filename and frontmatter date to today I can get it merged for you.

Also:

* consider adding `authorIsRacker: true` to the frontmatter to highlight the fact that this post was written by a Racker.
* consider adding `authorAvatar: <urlToYourGravatarHash>` to the frontmatter to display an avatar with your post instead of the default avatar. You'll need to [generate the Gravatar hash](https://en.gravatar.com/site/implement/hash/) on your own.

**Comment by ncoble** [comment]
@ktbartholomew  Thanks for getting back to me.  Changes are committed. I will get the gravatar setup for the next post.  Thanks!
-Nate

**Pull request merged**